### PR TITLE
Transform

### PR DESCRIPTION
**Pull request by moshez**

### CODE REVIEW

File: src/gather/api.py [hunk 1]
```diff
@@ -115,4 +120,18 @@ def run(argv, commands, version, output):
         return
     commands[argv[0]](argv)
 
-__all__ = ['Collector', 'run']
+def pair_with(second_element):
```
Comment (markrwilliams) [inline src/gather/api.py]: Why is this special enough to special case?

File: src/gather/api.py [hunk 2]
```diff
@@ -40,18 +40,23 @@ def _get_modules():
...
     """
 
-    def register(self, name=None):
+    name = attr.ib(default=None)
+    depth = attr.ib(default=1)
+
+    def register(self, name=None, transform=lambda x: x):
         """Register
 
+        :param name: optional. Name to register as (default is name of object)
```
Comment (markrwilliams) [inline src/gather/api.py]: :+1: for the docstring!

File: src/gather/api.py (hunk 1 above)
Comment (moshez) [inline src/gather/api.py]: As per off-line discussion, changed this to an explicit wrapper from a 2-tuple.

File: src/gather/api.py [hunk 3]
```diff
@@ -115,4 +120,27 @@ def run(argv, commands, version, output):
         return
     commands[argv[0]](argv)
 
-__all__ = ['Collector', 'run']
+@attr.s(frozen=True)
+class Wrapper(object):
```
Comment (markrwilliams) [inline src/gather/api.py]: :+1:

**Pull request merged**